### PR TITLE
add shell commands to build on Debian Stretch

### PR DESCRIPTION
**Pull request by rich-murphey**
For those considering trying to contribute, it might be helpful to help them see an easy path to building the sources.

### CODE REVIEW

**Comment by jwise** [comment]
Sounds great!  I didn't realize that this worked without the Pebble SDK.  @rich-murphey , can you confirm that you didn't need anything in your `localconfig.mk` -- just run those commands, and it builds?

**Comment by rich-murphey** [comment]
Yes, that's correct, I did a git clone and then build without creating a localconfig.mk.

That said, yes you're right it only builds a portion.  There are indeed portions that don't compile without the SDK, and I've got a recipe for that as well. I'll send another pull request with those.

Meanwhile, I ran out of time looking at extracting the FPGA blob.  I'd like to boil that down to shell commands as well if possible.

Anyway, I'll send another pull request with the SDK install commands.

**Comment by jwise** [comment]
FPGA blob extraction I think requires a hacked QEMU (to dump out the SPI packets) -- or reversing how it's compressed in the stock firmware.  (I don't object to someone doing that, but it will be somewhat less trivial!)  For right now, the best way to get a FPGA blob is to grab it from the link in the Discord.

File: README.md [hunk 1]
```diff
@@ -30,6 +30,29 @@ For example, the following bulids RebbleOS on Debian Stretch:
     cd FreeRTOS-Pebble
     make
 
+The Pebble SDK is a prerequisite for portions of RebbleOS. As an
+example, on Debian, one may create an installation directory for
+hte SDK using the following.
```
Comment (jwise) [inline README.md]: s/hte/the/

**Comment by rich-murphey** [comment]
Please feel free to edit anything or even rearrange the text as you see fit.  I almost always rewrite mulitple times before I think it's clear.  So feel free.

File: README.md [hunk 2]
```diff
@@ -30,6 +30,29 @@ For example, the following bulids RebbleOS on Debian Stretch:
...
+    tar -jxf pebble-sdk-4.5-linux64.tar.bz2
+    echo 'export PATH=~/pebble-dev/pebble-sdk-4.5-linux64/bin:$PATH' >> ~/.bash_profile
+    . ~/.bash_profile
+    sudo apt-get install python-pip python2.7-dev python-gevent
+    sudo pip install virtualenv
+    cd ~/pebble-dev/pebble-sdk-4.5-linux64
+    virtualenv --no-site-packages .env
+    source .env/bin/activate
+    pip install -r requirements.txt
+    deactivate
```
Comment (jwise) [inline README.md]: Adding a hyperlink to the Pebble SDK install page (so OS X users can vary it appropriately, etcetera) would be good, too.

**Comment by ginge** [comment]
Great work!

Be careful with the FPGA extraction, it's compressed in the actual binary.
We had to dump through an emulator.... If you can figure the compression
(it's likely not hard) it can be dumped cleanly!

:)

Thanks again

Barry Carter

**Comment by rich-murphey** [comment]
I ran out of time to look at this, but  the gist would be something like
git clone https://github.com/pebble/qemu.git
sed -i '.. edit code...'  hw/display/pebble_snowy_display.c
sed -i ps_display_determine_command_set to add code to dump the fpga.bin
run qemu

**Comment by rich-murphey** [comment]
Ah, so it's probably possible to extract the compressed blob as well?
If there are any pointers to where it is located, I'll be glad to take a look.
Otherwise, scripting qemu to dump it will at least lower the hurdle.

**Comment by jwise** [comment]
Yeah, that could work, too.  I think with the two fixes above, this would be great!  Could you add a commit to fix those two?  (I'm not at a real terminal to do it here, or I'd do it myself.)  Then I'd be happy to merge!

**Comment by jwise** [comment]
I don't think we've done much research for how it's stored in the official flash image.  I have a .idb lying around for some of the Pebble firmware, and I think @ginge does too; if you were excited about setting a breakpoint on write to the FPGA SPI in the official firmware and stepping around, it would not be the end of the world to find that control path in the .idb.

**Comment by ginge** [comment]
I have a script that does this:

dd if=tintin_fw.bin of=fpga.bin bs=1 count=30774 skip=830249

I can't verify that right now, but it looks legit

The format of the compressed FPGA is something like this

0xFF 0x00 (or the other way around) | ..Lattice... | compressed part

etc. You can compare with an existing one if you need the URL we can get it.

The qemu dump script is a good start I think. Clunky, but given the
alternative....

It's just occurred that those pebble-utils might be able to do something
with that compression

**Comment by rich-murphey** [comment]
I did a spell check after the 3rd commit.  Let me know if there is anything else I can do.

**Comment by jwise** [comment]
This is great, and a good way to help people get started with RebbleOS!  Thanks again.  If you're excited about developing RebbleOS further, you also might want to join us on Discord!

**Comment by ginge** [comment]
Thank you! Documentation is the hardest, and the additions are very much
welcome in helping get people into RebbleOS. Do you have a handle on
Discord? #fireware is where you will see us (ginge) if not.

Cheers!

**Comment by ginge** [comment]
Good grief, apologies.   #firmware

**Comment by rich-murphey** [comment]
Try to decompress the blob, I'll at least need a valid copy to verify against.  I can use qemu to get that if needed.  Anyway, it may be a while before I can return to that and I'm about to be away from the keyboard, so I'll catch you on Doscord later, perhaps tomorrrow.  

Thanks very much for the encouragement!!!!

**Pull request merged**